### ISSUE TITLE
Feature request: conditional nested sub-filters

### ISSUE THREAD

**Issue by rauanmayemir** [spiral/filters#7]
Currently, if a request has a lot of conditional nested fields, mapping and validating them can get very messy or straight up impossible.

The proposed idea is having dependent variants in their own filters and map them to the value 'statically'. Here's an example gist for pseudo schema:

```
public const SCHEMA = [
    'type' => 'data:type',
    'element' => [
        'nested' => 'data:type',
        'mapping' => [
            'type1' => ElementType1Filter::class,
            'type2' => ElementType2Filter::class,
        ]
    ],
];
```

So if the `type` equals `type1`, filter's `element` field will be an instance of `ElementType1Filter` and all the validation related to `type1` will happen inside `ElementType1Filter`.

**Comment by wolfy-j** [comment]
We can address with function-driven schema definition. I have about 7 people asking to improve filters, so I guess it's decided that we have to drastically update them.

**Comment by rauanmayemir** [comment]
Function-driven would work, but I thought the whole point with using constant schemas was performance.

**Comment by wolfy-j** [comment]
Not really, constants are being read and cached in memory between requests. So we can read the schema from any source, pretty much how cycle does.
